matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/state/StateService.kt

```diff
@@ -66,6 +66,11 @@ interface StateService {
      */
     suspend fun deleteAvatar()
 
+    /**
+     * Pin a message of the room.
+     */
+    suspend fun pinMessage(eventIds: MutableList<String>)
```

Comment:
This is taking a list of eventIds, so the doc is not accurate.

the SDK could expose a higher level API like

```kotlin
suspend fun pinEvent(eventId: String)
suspend fun unpinEvent(eventId: String)
```

and compute the new event data regarding the current one.

it may expose this API

```kotlin
suspend fun setPinnedEvents(eventIds: List<String>)
```

as a bonus for advanced usages like removing all the pinned events, and pinning multiple events in one request.

For the current usages added by this PR, I would just expose `pinEvent` and `unpinEvent` methods here.